go-controller/pkg/util/multi_network.go

```diff
@@ -158,6 +158,42 @@ func compareSubnetsString(subnetsString, newSubnetsString string) bool {
 	return true
 }
 
+// parseSubnetsString parses comma-seperated subnet string and returns the list of subnets
+func parseSubnetsString(clusterSubnetString string) ([]*net.IPNet, error) {
+	var subnetList []*net.IPNet
+	subnetStringList := strings.Split(clusterSubnetString, ",")
+
+	for _, subnetString := range subnetStringList {
+		subnetString = strings.TrimSpace(subnetString)
+		_, subnet, err := net.ParseCIDR(subnetString)
+		if err != nil {
+			return nil, err
+		}
+
+		subnetList = append(subnetList, subnet)
+	}
+
+	if len(subnetList) == 0 {
+		return nil, fmt.Errorf("failed to parse any subnets from %q", clusterSubnetString)
+	}
+
+	return subnetList, nil
+}
+
+// containsCIDR returns true if ipnet1 contains ipnet2
+func containsCIDR(ipnet1, ipnet2 *net.IPNet) bool {
+	mask1, _ := ipnet1.Mask.Size()
+	mask2, _ := ipnet2.Mask.Size()
+	return mask1 <= mask2 && ipnet1.Contains(ipnet2.IP)
+}
```

Comment:
This looks like something generally useful that could reside in `pkg/util/net.go` ?

Also it would probably be straight forward enough that some unit tests for our peace of mind?

Reply:
I agree; some unit tests would be helpful, and seem straight-forward.